sos/report/plugins/microk8s.py

```diff
@@ -44,6 +61,7 @@ def setup(self):
         self.add_cmd_output([
             f"microk8s {subcmd}" for subcmd in microk8s_subcmds
         ])
+        super(UbuntuKubernetes, self).setup()
```

Comment:
I've not seen this being done before within `sos` to call out a class outside. I guess if it works, then great. But would be good to get review from @TurboTurtle or @pmoravec on this

Reply:
Hmmmm. Yeah, that's not something we've done and also not something that `Plugin` is designed with the idea of doing.

I think what our typical design would be in this case would be to have a check within the base `kubernetes` class for the installation of microk8s, and if so add the microk8s specific bits there, allowing the rest of the plugin to focus on the shared content between the two. E.G.:

```python
class kubernetes(Plugin)

self.kube_cmd = 'kubectl'

if self.is_installed('microk8s'):
    self.kube_cmd = 'microk8s kubectl'
    self.add_copy_spec(
            "/var/snap/microk8s/current/credentials/client.config"
        )
    microk8s_subcmds = [<whatever isn't shared with kuberentes>]

    self.add_cmd_output([
        f"microk8s {subcmd}" for subcmd in microk8s_subcmds
    ])

[...the rest of the existing kubernetes plugin...]
```

I think this wasn't considered when the `microk8s` plugin was first written due to a significant lack of certain kubernetes bits that we're present in microk8s. If that's changed, I think the above is probably the correct choice (folding microk8s into kubernetes). If that _isn't_ the case, then I'd ask what we gain by calling `super(UbuntuKubernetes, self).setup()` here, since we'd either be getting not-a-lot more collection, or a whole ton of error output, depending on how `microk8s kubectl` behaves when trying to retrieve a resource that doesn't exist in that project.

Reply:
I think that makes sense, that we just added the `microk8s` bits 

I was already looking at the Kubeflow stuff internally, and actually we can already utilise the kubernetes plugin for his ,and having the microk8s bits within the kubernetes plugin would actually make it easier.

Something on the line of https://paste.ubuntu.com/p/gcrqHkNHxN/ would potentially work I think

Reply:
Sorry for the late replies :)

I think this all makes sense, the intent of microk8s is to gather the microk8s bits (microk8s status, microk8s config, etc...)
For the kubernetes pod logs we should indeed add the bits in the kubernetes plugin to gather the logs